prefect_hightouch/syncs/flows.py

```diff
@@ -0,0 +1,199 @@
+        max_wait_seconds: Maximum number of seconds to wait for the entire
+            flow to complete.
+        poll_frequency_seconds: Number of seconds to wait in between checks for
+            run completion.
+
+    Returns:
+        api_models.sync.Sync:
+        - `id: str`
+        - `slug: str`
+        - `workspace_id: str`
+        - `created_at: str`
+        - `updated_at: str`
+        - `destination_id: str`
+        - `model_id: str`
+        - `configuration: Dict`
+        - `schedule: Dict`
+        - `status: "models.SyncStatus"`
+        - `disabled: bool`
+        - `last_run_at: str`
+        - `referenced_columns: List[str]`
+        - `primary_key: str`
+
+    Examples:
+        Trigger a Hightouch sync run and wait for completion as a stand alone flow.
+        ```python
+        import asyncio
+
+        from prefect_hightouch import HightouchCredentials
+        from prefect_hightouch.syncs import trigger_sync_run_and_wait_for_completion
+
+        asyncio.run(
+            trigger_sync_run_and_wait_for_completion(
+                hightouch_credentials=HightouchCredentials(
+                    token="1abc0d23-1234-1a2b-abc3-12ab456c7d8e"
+                ),
+                sync_id=12345,
+                full_resync=True,
+                max_wait_seconds=1800,
+                poll_frequency_seconds=5,
+            )
+        )
+        ```
+
+        Trigger a Hightouch sync run and wait for completion as a subflow.
+        ```python
+        from prefect import flow
+
+        from prefect_hightouch import HightouchCredentials
+        from prefect_hightouch.syncs import trigger_sync_run_and_wait_for_completion
+
+        @flow
+        def sync_flow():
+            hightouch_credentials = HightouchCredentials.load("hightouch-token")
+            sync_metadata = trigger_sync_run_and_wait_for_completion(
+                hightouch_credentials=hightouch_credentials,
+                sync_id=12345,
+                full_resync=True,
+                max_wait_seconds=1800,
+                poll_frequency_seconds=10,
+            )
+            return sync_metadata
+
+        sync_flow()
+        ```
+    """
+    json_body = TriggerRunInput(full_resync=full_resync)
+    sync_run_future = await trigger_run.submit(
+        hightouch_credentials=hightouch_credentials,
+        sync_id=sync_id,
+        json_body=json_body,
+    )
+    await sync_run_future.result()
+
+    sync_status, sync_metadata = await wait_for_sync_run_completion(
+        hightouch_credentials=hightouch_credentials,
+        sync_id=sync_id,
+        max_wait_seconds=max_wait_seconds,
+        poll_frequency_seconds=poll_frequency_seconds,
+    )
+
+    if sync_status == SyncStatus.SUCCESS:
+        return sync_metadata
+    else:
+        raise TERMINAL_STATUS_EXCEPTIONS[sync_status](
+            f"Sync ({sync_metadata.slug!r}, ID {sync_id!r}) "
+            f"was unsuccessful with {sync_status.value!r} status"
+        )
```

Comment:
Can we use `.get()` here and raise a `HightouchSyncRunError` if we get a unexpected status back?